### PR TITLE
Deploy to gh pages

### PR DESCRIPTION
**Pull request by elias-pap**
`yarn deploy` will create the static files and push them to `gh-pages` branch of this repo. This will be picked up by GH pages and the docs will be deployed.

For the custom domain to work, the following DNS CNAME record should be set up: `docs.etebase.com` --> `etesync.github.io`

### CODE REVIEW

**Comment by tasn** [comment]
I think we can probably also just add a github action:
```
name: Build and Deploy
on:
  push:
    branches:
      - master
jobs:
  build-and-deploy:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v2.3.1

      - name: Install and Build 🔧
        run: |
          yarn
          yarn build

      - name: Deploy 🚀
        uses: JamesIves/github-pages-deploy-action@4.1.3
        with:
          branch: gh-pages
          folder: build
```
I think this should work (I've done something similar before).

**Comment by elias-pap** [comment]
> I think we can probably also just add a github action:
> 
> ```
> name: Build and Deploy
> on:
>   push:
>     branches:
>       - master
> jobs:
>   build-and-deploy:
>     runs-on: ubuntu-latest
>     steps:
>       - name: Checkout
>         uses: actions/checkout@v2.3.1
> 
>       - name: Install and Build 🔧
>         run: |
>           yarn
>           yarn build
> 
>       - name: Deploy 🚀
>         uses: JamesIves/github-pages-deploy-action@4.1.3
>         with:
>           branch: gh-pages
>           folder: build
> ```
> 
> I think this should work (I've done something similar before).

Addressed in 6d216c1, based on [these suggestions](https://docusaurus.io/docs/deployment#triggering-deployment-with-github-actions). It uses a similar GH action for deployment (peaceiris/actions-gh-pages@v3).

File: package.json [hunk 1]
```diff
@@ -6,14 +6,14 @@
     "start": "docusaurus start",
     "build": "docusaurus build",
     "swizzle": "docusaurus swizzle",
-    "deploy": "docusaurus deploy"
+    "deploy": "GIT_USER=tasn docusaurus deploy"
```
Comment (tasn) [inline package.json:9]:
```suggestion
    "deploy": "docusaurus deploy"
```

**Comment by tasn** [comment]
Made one more comment and then we are ready to go!

File: package.json (hunk 1 above)
Comment (tasn) [reply package.json]: Wouldn't it mess up the branch deployment though? Also, I can set it manually locally if I ever want that.

**Comment by tasn** [comment]
Yay, thanks!

**Pull request merged**